ui/src/features/WorkspaceSettings/components/GeneralSettings.tsx

```diff
@@ -6,25 +6,44 @@ import { useWorkspace } from "@flow/lib/gql";
 import { useT } from "@flow/lib/i18n";
 import { useCurrentWorkspace } from "@flow/stores";
 
+type Errors = "delete" | "update";
+
 const GeneralSettings: React.FC = () => {
   const t = useT();
-  const [currentWorkspace] = useCurrentWorkspace();
-  const { deleteWorkspace } = useWorkspace();
+  const [currentWorkspace, setCurrentWorkspace] = useCurrentWorkspace();
+  const { deleteWorkspace, updateWorkspace } = useWorkspace();
   const navigate = useNavigate();
-  const [showError, setShowError] = useState(false);
+  const [showError, setShowError] = useState<Errors | undefined>(undefined);
+  const [workspaceName, setWorkspaceName] = useState(currentWorkspace?.name);
+  const [loading, setLoading] = useState(false);
 
   const handleDeleteWorkspace = async () => {
-    setShowError(false);
+    setLoading(true);
+    setShowError(undefined);
     if (!currentWorkspace) return;
-    // TODO: this trigger a pop up for confirming
+    // TODO: this should trigger a pop up for confirming
     const { workspaceId } = await deleteWorkspace(currentWorkspace.id);
 
     if (!workspaceId) {
-      setShowError(true);
+      setShowError("delete");
       return;
     }
     navigate({ to: "/workspace" });
   };
+
+  const handleUpdateWorkspace = async () => {
+    setLoading(true);
+    setShowError(undefined);
+    if (!currentWorkspace?.id || !workspaceName) return;
+    const { workspace } = await updateWorkspace(currentWorkspace?.id, workspaceName);
+    setLoading(false);
+    if (!workspace) {
+      setShowError("update");
+      return;
+    }
+    // TODO: Not sure if this is correct to set here
+    setCurrentWorkspace(workspace);
```

Comment:
I feel like this shouldnt be necessary if we are using graphql and the cache effectively. 
I feel like updating the workspace should automatically update the cache, and thus update wherever the workspace(s) query is used.
If that isn't happening, perhaps we need to either
- update the useQueries code to update the cache specifically, OR
- set up the cache better through more specific queryKeys

Reply:
the cache updates okay but workspace doesn't reference cache, it copies the workspace from it. So even after updating the cache, it doesn't show up in the `currentWorkspace`

Reply:
Does this apply even if you use useCallback/useMemo? 🤔

Reply:
I haven't used `useCallback/useMemo` in reference to setCurrentWorkspace and currentWorkspace , let me try this.

Reply:
![image](https://github.com/reearth/reearth-flow/assets/20891087/77754ded-5d67-4a33-a8ac-83112905293a)
this fixed it, since id wasn't the same it didn't set the currentworkspace again